proof/crefine/ARM/Ipc_C.thy

```diff
@@ -2951,9 +2951,10 @@ proof -
                    del: Collect_const)
        apply csymbr
        apply (rename_tac "lngth")
-       apply (simp add: mi_from_H_def mapME_def del: Collect_const cong: bind_apply_cong)
+       apply (unfold mapME_def)[1]
+       apply (simp add: mi_from_H_def del: Collect_const)
```

Comment:
This was a bit annoying, in eta-contracted form the simplifier does not rewrite `mapME` inside of the abstract program. It's very possible that there is a nicer way of doing this but for a first draft I just went with `unfold`.

Reply:
simp with a cong rule?

Reply:
I tried a couple, including a new one that I thought might help, but couldn't get anything to apply. I might try again now that I've finished a first pass through the proofs.

Reply:
I'm fine with the unfold here. Compared to the benefit of much fewer `rv` and `xa` etc, that is a small price to pay.